### PR TITLE
fix: bind NotificationManager as Facade for Octane

### LINKED ISSUE #2569: NotificationManager don't cleanup in Octane
### Package

filament/filament

### Package Version

v2.12.17

### Laravel Version

v9.14.1

### Livewire Version

v2.10.5

### PHP Version

8.0.16

### Bug description

Filament\NotificationManager uses a static variable '$notifications' that is not flushed between requests, causing the same notification to show again.

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by TFSThiagoBR98**
Fix #2569

### CODE REVIEW

**Review by ryangjchandler — COMMENTED** [review]
This doesn't seem right... we're binding something to the container by resolving it from the container?

File: packages/admin/src/Facades/FilamentNotificationManager.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use Livewire\Response;
+use Illuminate\Support\Facades\Facade;
+
+/**
+ * @method static void notify(string $status, string $message)
+ * @method static Response handleLivewireResponses(Component $component, Response $response)
+ *
+ * @see NotificationManager
+ */
+class FilamentNotificationManager extends Facade
```
Comment (ryangjchandler) [inline packages/admin/src/Facades/FilamentNotificationManager.php:16]: The class is namespaced, no need for `Filament` in the class name.

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: packages/admin/src/Facades/FilamentNotificationManager.php (hunk 1 above)
Comment (TFSThiagoBR98) [reply packages/admin/src/Facades/FilamentNotificationManager.php]: Fixed

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: packages/admin/src/FilamentServiceProvider.php [hunk 2]
```diff
@@ -82,6 +83,10 @@ public function packageRegistered(): void
             return app(FilamentManager::class);
         });
 
+        $this->app->scoped('filament_notification_manager', function (): NotificationManager {
```
[the remark this answers is not in the document]
Comment (TFSThiagoBR98) [reply packages/admin/src/FilamentServiceProvider.php]: Done

File: packages/admin/src/FilamentServiceProvider.php [hunk 3]
```diff
@@ -82,6 +83,10 @@ public function packageRegistered(): void
...
+            return app(NotificationManager::class);
```
[the remark this answers is not in the document]
Comment (TFSThiagoBR98) [reply packages/admin/src/FilamentServiceProvider.php]: Done

**Comment by ryangjchandler** [comment]
@TFSThiagoBR98 There's no need to re-write your commits and force push etc. If a PR has a tonne of commits, we just squash it anyway.

**Review by ryangjchandler — APPROVED** [review]

**Pull request merged**